swarm/swarm.go

```diff
@@ -102,29 +105,26 @@ func NewSwarm(ctx *node.ServiceContext, backend chequebook.Backend, ensClient *e
 	// setup local store
 	log.Debug(fmt.Sprintf("Set up local storage"))
 
-	self.dbAccess = network.NewDbAccess(self.lstore)
-	log.Debug(fmt.Sprintf("Set up local db access (iterator/counter)"))
-
-	// set up the kademlia hive
-	self.hive = network.NewHive(
-		common.HexToHash(self.config.BzzKey), // key to hive (kademlia base address)
-		config.HiveParams,                    // configuration parameters
-		swapEnabled,                          // SWAP enabled
-		syncEnabled,                          // syncronisation enabled
+	kp := network.NewKadParams()
+	to := network.NewKademlia(
+		common.FromHex(config.BzzKey),
+		kp,
 	)
-	log.Debug(fmt.Sprintf("Set up swarm network with Kademlia hive"))
 
-	// setup cloud storage backend
-	self.cloud = network.NewForwarder(self.hive)
-	log.Debug(fmt.Sprintf("-> set swarm forwarder as cloud storage backend"))
+	config.HiveParams.Discovery = true
```

Comment:
In fact, here we are using `config.HiveParmas`, so I wonder why `HiveParams.Init()` is commented?